bf2fc6cc711aee1a0c2a/kas-fleet-manager
# Put in place config and code change to allow the midstream pipelines to run against RH SSO as the SSO provider

## Description


Closes https://issues.redhat.com/browse/MGDSTRM-7819

This allow to run the midstream CI using RH SSO as the provider without leaking service accounts and with an even reduced load to the SSO service thanks to https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1355.

See the linked documentation in https://issues.redhat.com/browse/MGDSTRM-7819 for further background on this work. 

## Verification Steps

Run the `pr_chech.sh` file providing it with the needed credentials:
To successfully run that script, the following has to be done:
1. Change https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/dd5b40ab62cad808f26b6fcd39aeea15e5579751/docker/Dockerfile_template_mocked#L1 and https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/dd5b40ab62cad808f26b6fcd39aeea15e5579751/docker/Dockerfile_template#L1 to `FROM postgres:13`
2. Make sure you've the needed credentials. See the pr_check.sh for the needed environment variables. The env var in  https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/dd5b40ab62cad808f26b6fcd39aeea15e5579751/pr_check.sh#L58-L61 are not so mandatory for this verification as they are only needed to publish the test output to the test portal, so dummy values can be used.
3. Run the script with first, `OCM_ENV` set to `integration` and then `OCM_ENV` set to `development` so that we are sure that each mode works as expected.

## Checklist (Definition of Done)
